keypair/src/lib.rs

```diff
@@ -34,8 +33,8 @@ impl Keypair {
     /// Constructs a new, random `Keypair` using `OsRng`
     #[allow(clippy::new_without_default)]
     pub fn new() -> Self {
-        let mut rng = OsRng;
-        Self(ed25519_dalek::SigningKey::generate(&mut rng))
+        let secret_bytes = rand::random::<[u8; Self::SECRET_KEY_LENGTH]>();
```

Comment:
Technically, this changes from explicitly using OsRng to using ThreadRng. Not sure it matters, just pointing it out. It is probably for the best.